lib/streamlit/runtime/state/query_params.py

```diff
@@ -88,6 +91,13 @@ def __delitem__(self, key: str) -> None:
         except KeyError:
             raise KeyError(missing_key_error_message(key))
 
+    def update(self, mapping: Mapping[str, str]):
+        # We define a custom update function that only sends the message once
```

Comment:
super nit:`message` should likely be specified to `forwardMsg` or `forward message`